web/client/codechecker_client/cmd_line_client.py

```diff
@@ -958,14 +967,136 @@ def report_to_html(client, reports, output_dir):
         html_builder.create_index_html(output_dir)
         print_stats(reports, file_stats, severity_stats)
 
-    def print_reports(client, reports, output_format):
+    def get_report_show_info(report):
+        """ Get information from report which will be used during print. """
+        if isinstance(report, Report):
+            # Report is coming from a plist file.
+            bug_line = report.main['location']['line']
+            bug_col = report.main['location']['col']
+            check_name = report.main['check_name']
+            sev = context.severity_map.get(check_name)
+            file_name = report.main['location']['file_name']
+            checked_file = file_name \
+                + ':' + str(bug_line) + ":" + str(bug_col)
+            check_msg = report.main['description']
+            source_line =\
+                get_line_from_file(report.main['location']['file_name'],
+                                   bug_line)
+        else:
+            # Report is coming from CodeChecker server.
+            bug_line = report.line
+            bug_col = report.column
+            check_name = report.checkerId
+            sev = ttypes.Severity._VALUES_TO_NAMES[report.severity]
+            file_name = report.checkedFile
+            checked_file = file_name
+            if bug_line is not None:
+                checked_file += ':' + str(bug_line) + ":" + str(bug_col)
+            check_msg = report.checkerMsg
+            if lines_in_files_requested:
+                source_line_contents = client.getLinesInSourceFileContents(
+                    lines_in_files_requested, ttypes.Encoding.BASE64)
+                source_line = base64.b64decode(
+                    source_line_contents[report.fileId][bug_line])
+        return bug_line, bug_col, sev, file_name, checked_file, check_name, \
+            check_msg, source_line
+
+    def get_gerrit_review_changed_files(changed_file_path):
+        """ Return a list of changed files.
+
+        Process the given gerrit changed file object and return a list of
+        file paths which changed.
+        """
+        ret = []
+
+        if not changed_file_path:
+            return ret
+
+        with open(changed_file_path) as changed_file:
+            # File is likely to contain some garbage values at start,
+            # only the corresponding json should be parsed.
+            json_pattern = re.compile(r"^\{.*\}")
+            for line in changed_file.readlines():
+                if not re.match(json_pattern, line):
+                    continue
```

Comment:
Is there a guarantee, that the whole json string will be in one line? I think it would be more robust to read the whole file and use regex.search to match the outermost json object (you could use the same regex).